### ISSUE TITLE
Fix false positives for non-CSS files in no-empty-source

### ISSUE THREAD

**Issue by shinnn** [stylelint/stylelint#3233]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug in combination of [`no-empty-source`](https://github.com/stylelint/stylelint/blob/9.1.3/lib/rules/no-empty-source/README.md) rule and non-CSS files for example HTML and Markdown

> Which rule, if any, is this issue related to?

`no-empty-source`

> What CSS is needed to reproduce this issue?

An HTML, Markdown or Vue file with only white spaces.

> What stylelint configuration is needed to reproduce this issue?

e.g.

```js
module.exports = {
  rules: {
    'no-empty-source': true
  }
};
```

> Which version of stylelint are you using?

e.g. `9.1.3`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

API, but probably the problem is reproducible with any interface.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No.

> What did you expect to happen?

```js
'use strict';

const {lint} = require('stylelint'); // v9.1.3

(async () => {
  const {results} = await lint({
    code: '',
    codeFilename: 'empty.html',
    config: {
      rules: {
        'no-empty-source': true
      }
    }
  });

  console.log(results[0].warnings[0]);
})();
```

When I run this code, I expect the result should contain no warnings about `no-empty-source`.

> What actually happened (e.g. what warnings or errors you are getting)?

The code above prints the following:

```js
{ line: 1,
  column: 1,
  rule: 'no-empty-source',
  severity: 'error',
  text: 'Unexpected empty source (no-empty-source)' }
```

This warning seems valid at first glance, because `example.html` is actually a zero-byte file and `no-empty-source` warning looks appropriate for it. However, as the file is not CSS and has no `<style>` tags, I think stylelint should not report any errors about it.

If users really want to invalidate empty HTML, they can use a HTML linter separately and wouldn't expect stylelint to warn about it.

**Comment by jeddy3** [comment]
> However, as the file is not CSS and has no <style> tags, I think stylelint should not report any errors about it.

I agree that the behaviour should be this.

@gucong3000 Do you think this needs addressing in stylelint itself or postcss-html?

**Comment by gucong3000** [comment]
> Do you think this needs addressing in stylelint itself or postcss-html?
May it is work: https://github.com/stylelint/stylelint/blob/c425840d2d40558b7b021459345143d3e58f1f48/lib/utils/eachRoot.js#L1-L22
